### PR TITLE
Loop refactor and blocking improvement

### LINKED ISSUE #12: Dispatch blocks, incorrect async
After unpacking, the reading loop will [`yield from`](https://github.com/numberoverzero/bottom/blob/master/bottom/connection.py#L50) the event dispatch, which means it waits until all handlers run for the event, before continuing.

Instead, `events.trigger` should be a non-blocking call.  That is, it should push the event into a queue, then return execution to the caller.

---

I'm working on [accordian](https://github.com/numberoverzero/accordian) to fix this - however, in trying to use the 3.5 syntax (`await` and `async` as keywords) I've run into other problems.  Once I've finished testing the library, I'll integrate it into bottom.  This will require 3.5+ to use bottom, but the migration from 3.4 (current requirement) to 3.5 is virtually painless (I haven't seen any reports of problems migrating).

### LINKED ISSUE #13: Tests use default event loop
Similar to [accordian](https://github.com/numberoverzero/accordian/blob/master/tests/conftest.py#L6) we should be creating a new event loop per test.

This will be a larger part of the loop refactor that needs to occur for the main components: Client and Connection.

### PR DESCRIPTION
**Pull request by numberoverzero**
Building on the work in #16 this resolves #12 and resolves #13.

The fix for blocking during dispatch turned out to be remarkably easier than the proposed work in #12, as @AMorporkian pointed out in #16.  By making `EventsMixin.trigger` simply enqueue the callbacks into the event loop (turns out async is hard) there's no blocking and everything gets better.

Because this changes the semantics of `.trigger` (a significant portion of the api) this will include a bump to 1.0.0

### CODE REVIEW

**Comment by AMorporkian** [comment]
If we're bumping it up to 3.5, would it be worth using async/await as opposed to `@asyncio.coroutine`/`yield from`?

**Comment by numberoverzero** [comment]
Yep, good call.  It's a weird story how we got to 3.5 in tox, involving travis only having 3.4.2 which caused test failures.  Will migrate.

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2014 Joe Cross
+Copyright (c) 2016 Joe Cross
```
Comment (meshy) [inline LICENSE]: I suspect you prpobably [don't need to update this](http://stackoverflow.com/a/2391555/400691).

Comment (numberoverzero) [inline LICENSE]: TIL!  Reverted, thanks.

Comment (meshy) [inline LICENSE]: Welcome `:)`

**Comment by numberoverzero** [comment]
@AMorporkian any more feedback?  I'd like to get the other branch up for PR and keep working on the protcol refactor, which will simplify the interface substantially.

I don't want to kick those off until this is fixed and released for existing users.  It's a pretty big fix.

**Comment by AMorporkian** [comment]
I spent some time looking over the code. It all looks good on my end. I have some thoughts on API improvements, but it's stuff that I think could (and probably should) wait for 2.0.0.

One stylistic thing though. I wasn't going to bring it up, and please don't think this is me bikeshedding. By PEP8, docstrings should use `"""`, not `'''`. It's a super minor thing, and if you feel strongly about it please do ignore me.

**Comment by numberoverzero** [comment]
Haha, I've been migrating the strings to `"""` as I find them, actually.  Probably missed a few, but I'm happy to merge replacements.  Not sure why my style switched halfway through last year.

I'm going to merge, feel free to open an issue to discuss API changes.

**Pull request merged**